lib/assignment/widgets/assignments_screen.dart

```diff
@@ -174,16 +176,16 @@ class AssignmentCard extends StatelessWidget {
 
     return <Widget>[
       if (assignment.courseId != null)
-        FancyCachedBuilder<Course>(
-          controller: assignment.courseId.controller,
-          builder: (_, course, __) {
+        EntityBuilder<Course>(
+          id: assignment.courseId,
+          builder: handleEdgeCases((context, course, fetch) {
```

Comment:
`CourseChip` already handles empty states.
```suggestion
          builder: handleError((context, course, fetch) {
```